drivers/flash/nrf_qspi_nor.c

```diff
@@ -534,19 +534,47 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 			  size_t size)
 {
+	const void *sptr = src;
+	size_t dlen = size;
+	u8_t buf[4];
+
 	if (!src) {
 		return -EINVAL;
 	}
 
-	/* write size must be non-zero multiple of 4 bytes */
-	if (((size % 4U) != 0) || (size == 0)) {
-		return -EINVAL;
-	}
 	/* address must be 4-byte aligned */
 	if ((addr % 4U) != 0) {
 		return -EINVAL;
 	}
 
+	/* read size must be non-zero multiple of 4 bytes */
+	if (size < 4U) {
+		src = buf;
+		size = sizeof(buf);
+		nrfx_err_t res = nrfx_qspi_read(buf, size, addr);
```

Comment:
Should we add a comment here stating why we perform a read in the write-function?

Reply:
Added comment